### ISSUE TITLE
YAML Configuration | Surface

### ISSUE THREAD

**Issue by danholdaway** [GEOS-ESM/swell#76]
## Agreement with GSI using GeoVaLs in UFO (Hofx)

- [ ] Done

## Agreement with GSI using GeoVaLs in UFO (Quality control)

- [ ] Done

## Agreement with GSI using Swell

- [ ] Done

## Current issues:

- None

**Comment by gmao-jjin3** [comment]
![corscat_qc_omb_vs_gsiombbc](https://user-images.githubusercontent.com/61251050/199300435-aa8d26bf-c7c1-45db-9949-78f5fa767382.png)
![corscat_eferr_vs_finaler](https://user-images.githubusercontent.com/61251050/199300490-cd42879d-d787-4ab4-b479-59de69cd0892.png)

These Sfc/surface_pressure figures are made with the EMC yaml 
https://github.com/NOAA-EMC/GDASApp/blob/feature/sfcps_oberrors/parm/atm/obs/testing/sfc.yaml
Except that "geometric height" are replaced by "geopotenial height". However, they actually are saved from the same values.  There is a related issue in https://github.com/GEOS-ESM/swell/issues/78 .

**Comment by gmao-jjin3** [comment]
![density_qc_omb_vs_gsiombbc](https://user-images.githubusercontent.com/61251050/208810035-f0b5c8ed-bf9d-464b-918e-6a9f69e58ccf.png)

These histograms of surface pressure O-F after BC and QC are identical in UFO and GSI.

**Comment by gmao-jjin3** [comment]
![corscat_eferr_x_GsiDup_vs_finaler](https://user-images.githubusercontent.com/61251050/208810608-5bcca055-ab3b-466c-b380-ea3aa516bafe.png)

This figure shows observation errors in UFO are the same as in GSI after these UFO values are multiplied by the saved GSI duplicate factors.  It suggests that error assignments are done correctly so far and those final observational errors are expectedt to match the GSI values once the UFO duplicate factor is done.

**Comment by gmao-jjin3** [comment]
![corscat_qc_omb_vs_gsiombbc](https://github.com/GEOS-ESM/swell/assets/61251050/abadab69-6a64-4200-89a1-fc1372b484ad)
Validation with x0044. Feb 8, 2024.

**Comment by gmao-jjin3** [comment]
Validation results from X0048 output.
![density_qc_omb_vs_gsiomb](https://github.com/GEOS-ESM/swell/assets/61251050/f396bfdb-c5b4-4697-b1e6-7dedc0e2684f)
![corscat_eferr_x_GsiDup_vs_finaler](https://github.com/GEOS-ESM/swell/assets/61251050/b57ba9fa-6492-4630-9b05-4dbf0af00f1a)
![corscat_eferr_vs_finalerr](https://github.com/GEOS-ESM/swell/assets/61251050/749525f3-6b47-4f85-9f1b-42fe26a702ad)
![corscat_qc_omb_vs_gsiombbc](https://github.com/GEOS-ESM/swell/assets/61251050/809b431f-4d7d-4d10-9131-c77194303a47)

![corscat_qc_omb_vs_hofx-GsiHofXBc](https://github.com/GEOS-ESM/swell/assets/61251050/9a7e7357-59d4-4618-bd53-15bcdef8cfeb)

**Comment by gmao-jjin3** [comment]
It turns out there is a bug in the UFO operator. UFO and GSI hofx values match after it is fixed in a new PR https://github.com/JCSDA-internal/ufo/pull/2987
![corscat_qc_omb_vs_hofx-GsiHofXBc](https://github.com/GEOS-ESM/swell/assets/61251050/24da5952-2a94-416e-beaf-6e3df0b85120)
